src/components/singleProductPage/SingleProduct.jsx

```diff
@@ -20,6 +26,45 @@ export default function SingleProduct({ productData }) {
     slidesToScroll: 1,
   };
 
+  const addItemToShoppingCart = () => {
+    let isItemAdded = false;
+    if (productsData.length !== 0) {
+      for (let i = 0; i < productsData.length; i++) {
+        let currentItemId = singleProductData.id;
+        if (productsData[i].id === currentItemId) {
+          productsData[i].quantity++;
+          isItemAdded = true;
+          break;
+        }
+      }
+      if (!isItemAdded) {
+        dispatch(addProductToCart(singleProductData));
+      }
+    }
+  };
+
+  const handleQuantitySubtraction = () => {
+    const copyOfData = [singleProductData].slice();
+    for (let i = 0; i < copyOfData.length; i++) {
+      const currentItemId = singleProductData.id;
+      if (copyOfData[i].id === currentItemId) {
+        copyOfData[i].quantity--;
+      }
+    }
+    dispatch(modifyProductQuantity(copyOfData));
```

Comment:
see how this is duplicated logic from the other page. remember what i said about copy-pasting code: any copy pasting is a sign that something is wrong; in this case you may need to think more generally about a cart. 

* redux has a cart
* redux exposes operations to modify a cart
* any page should be able to modify that cart very easily using only redux operations

imagine if you had like 8 different places you had to modify the cart, the current approach would make it extremely hard